From: Pascal Nowack <Pascal.Nowack@gmx.de>
Date: Tue, 17 Mar 2026 18:27:51 +0100
Subject: tests/rdp: Ignore cursor updates when checking for the screen update

Since the FreeRDP 1.x days, the EndPaint callback is also called when
non-screen-related updates happen. For gnome-remote-desktop, this
specifically means that cursor updates trigger the EndPaint callback of
the test client. Since cursor updates are not tied to frame updates in
gnome-remote-desktop, this can lead to a race whether the cursor update
happens first, or the frame update one. In the end, the order does not
matter, but the test client still needs to account for that, which is
currently not the case. And that leads to random test failures, where
the invalidation rect of size 1x1 happens at the position (0, 0). This
is the default rect.

To fix these random failures, simply check whether the damage region is
actually present.

Origin: https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/merge_requests/392
---
 tests/test-client-rdp.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/test-client-rdp.c b/tests/test-client-rdp.c
index 219e68d..a06a187 100644
--- a/tests/test-client-rdp.c
+++ b/tests/test-client-rdp.c
@@ -194,6 +194,9 @@ rdp_update_end_paint (rdpContext *rdp_context)
   int32_t x, y;
   uint32_t width, height;
 
+  if (rdp_context->gdi->primary->hdc->hwnd->invalid->null)
+    return TRUE;
+
   x = rdp_context->gdi->primary->hdc->hwnd->invalid->x;
   y = rdp_context->gdi->primary->hdc->hwnd->invalid->y;
   width = rdp_context->gdi->primary->hdc->hwnd->invalid->w;
